src/main/java/uk/gov/hmcts/divorce/solicitor/event/page/Applicant2ServiceDetails.java

```diff
@@ -94,4 +107,30 @@ public void addTo(final PageBuilder pageBuilder) {
                     "This address will be used to notify them about the application")
             .done();
     }
+
+    public AboutToStartOrSubmitResponse<CaseData, State> midEvent(CaseDetails<CaseData, State> details,
+                                                                  CaseDetails<CaseData, State> detailsBefore) {
+        CaseData caseData = details.getData();
+        Applicant applicant2 = caseData.getApplicant2();
+        List<String> validationErrors = new ArrayList<>();
+
+        boolean validApplicantEmail;
+        if (applicant2.getEmail() != null && !applicant2.getEmail().isEmpty()) {
```

Comment:
worth using `import static org.apache.commons.lang3.StringUtils.isNotEmpty;`

Reply:
Added this.